### ISSUE TITLE
[BUG] [CRITICAL-261] SKIPPED. Expected Quality not found.

### ISSUE THREAD

**Issue by unclechris4life** [nishad/udemy-dl-windows#139]
[BUG]
I have been using this project for a while now. However, on April 15, 2017, I noticed an issue.
When I execute the script and it attempts to get course data, it gives an error, CRITICAL-261 SKIPPED: Expected Quality not found.
![udemy](https://cloud.githubusercontent.com/assets/12344226/25061266/41e31944-21aa-11e7-9f80-de2d72601e17.JPG)
 
Version : **2.1.2**

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [X] I have read and understood the [contributors guide](https://github.com/nishad/udemy-dl-windows/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [X] I have checked that the issue I'm posting isn't already reported.
- [X] I have checked that the issue I'm porting isn't already solved and no duplicates exist in [closed tab](https://github.com/nishad/udemy-dl-windows/issues?q=is%3Aissue+is%3Aclosed)
- [X] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

**My familiarity with the project is as follows (check one, eg [X]):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [X] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

**If the project maintainer has any additional requirements, they will be listed here.**

- No additional requirements.

---

I have been using this project for a while now. However, on April 15, 2017, I noticed an issue.
When I execute the script and it attempts to get course data, it gives an error, CRITICAL-261 SKIPPED: Expected Quality not found.

This is how I executed the script
udemy-dl.exe -u *** -p *** -q 4321 -o c:\a --lecture-start 32 https://www.udemy.com/an-entire-mba-in-1-courseaward-winning-business-school-prof/learn/v4/content

I then tried specifying different video quality values but still got the same error. Here is the new code I executed.
udemy-dl.exe -u *** -p *** -q 4321 -o c:\a --lecture-start 32 https://www.udemy.com/an-entire-mba-in-1-courseaward-winning-business-school-prof/learn/v4/content

When this error occurs, all accompanying attachments are downloaded. Only the videos are not downloaded. I have attached a screenshot of my experience.

**Comment by farangkao** [comment]
I've got the same error, until i've noticed that aria2c was blocked by my firewall.
My Firewall usually popup's a message, but since the first part of the download (before the actual video download starts) takes quiet a long time ,i walked away from my computer ,so i didn't notice the message at first.

Maybe you have a similar problem?

**Comment by m0542128150** [comment]
I have the same error , I'm trying to download this course https://www.udemy.com/an-entire-mba-in-1-course-arabic-version-by-chris-haroun

The folders and Quizzes downloaded but the videos not downloaded , the error message is : CRITICAL-261] SKIPPED. Expected Quality not foun

Thanks for help..

**Comment by smeggingsmegger** [comment]
I have a potential fix for that here: https://github.com/nishad/udemy-dl-windows/pull/140

**Comment by JacquesvanWyk** [comment]
Having the same problem on windows.

**Comment by hvsh** [comment]
+1

**Comment by m0542128150** [comment]
Dear friends.. Any good news about this error

**Comment by zhedongzheng** [comment]
Same bug here

**Comment by jzarebski** [comment]
Ditto on the error over here as well using windows 7, I have installed python but that did not help either and tried @smeggingsmegger fix to no avail, although I may have made an error in applying the fix I wasn't 100% sure I did that right.

**Comment by toeknee581** [comment]
I was able to build the updated exe using the instructions I posted here: https://github.com/nishad/udemy-dl-windows/issues/143

**Comment by uckthis** [comment]
I see this issue every where, but I don't see a solution that a layman would understand. I don't know ABC of programming or scripts,. Just tell me in layman terms what do I do to download the course without getting this Critical 261 error? Link me to the updated file which I'd just replace to magically disappear the issue.
Thank you.

**Comment by darkeden82** [comment]
toeknee581  : your .exe seems to be invalid or corrupted,
I get an error everytime I try to execute it : udemy-dl.exe is not a valid win32 application

can you please check it ?

**Comment by tofanelli** [comment]
@darkeden82 .... because this isn't a w32 file... this must be run in a command prompt

**Comment by gokaybiz** [comment]
"install the patched version that works you should do:
sudo pip uninstall udemy-dl
sudo pip install udemy-dl==0.2.2a1
And then it works fine"

**Comment by tofanelli** [comment]
@gokaybiz .... updated the file, and still not working... and yes, i'm running in a elevated admin prompt, just to ensure i will not have any headaches regarding security 

what else can i do? =\

**Comment by gokaybiz** [comment]
I was tried on linux and worked very well. I'm sorry...

**Comment by arshahuja** [comment]
@JacquesvanWyk  @farangkao  @m0542128150  @smeggingsmegger 
udemy @ youtube vedios can be downloaded by this method also check this out!!
https://github.com/arshahuja/udemy-download-script

**Comment by tofanelli** [comment]
I used this one [Udemy-dl](https://github.com/r0oth3x49/udemy-dl) and worked flawlessly =)

**Comment by caglartoklu** [comment]
@tofanelli I have tried that one on Windows 10 / Python 3 after your comment and it worked flawlessly too.

**Comment by tofanelli** [comment]
@caglartoklu .... superb then.... glad you liked it and its working for you now =)

**Comment by nishad** [comment]
Fixed with v2.1.3 (https://github.com/nishad/udemy-dl-windows/releases/tag/v2.1.3)
